sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/Hotspots/DoNotHardcodeCredentials.cs

```diff
@@ -93,5 +95,47 @@ private class AssignmentExpressionBannedWordsFinder : CredentialWordsFinderBase<
                 syntaxNode.Left.IsKnownType(KnownType.System_String, semanticModel) &&
                 syntaxNode.Right.IsKind(SyntaxKind.StringLiteralExpression);
         }
+
+        private class StringLiteralBannedWordsFinder : CredentialWordsFinderBase<LiteralExpressionSyntax>
```

Comment:
could you add a high level comment about this class, as it's a special usage of the CredentialWordsFinderBase?
- it should clarify why `GetVariableName` returns `null`
- it should say that it looks inside strings for a specific pattern - what pattern?